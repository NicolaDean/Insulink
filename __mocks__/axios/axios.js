
const mockaxios = jest.fn(async(config) =>{
    console.log("CONFIG:" + JSON.stringify(config));
    return new Promise(resolve=>resolve()); //THIS ALLOW TO FAKE THE REQUEST
});

module.exports = mockaxios;