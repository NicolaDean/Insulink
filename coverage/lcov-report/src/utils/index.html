
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/utils</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> src/utils</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">51.97% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>184/354</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">20.16% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>24/119</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">50% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>37/74</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">52.02% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>180/346</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line medium'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="apiHelper.js"><a href="apiHelper.js.html">apiHelper.js</a></td>
	<td data-value="92.5" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 92%"></div><div class="cover-empty" style="width: 8%"></div></div>
	</td>
	<td data-value="92.5" class="pct high">92.5%</td>
	<td data-value="40" class="abs high">37/40</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="88.88" class="pct high">88.88%</td>
	<td data-value="9" class="abs high">8/9</td>
	<td data-value="92.5" class="pct high">92.5%</td>
	<td data-value="40" class="abs high">37/40</td>
	</tr>

<tr>
	<td class="file high" data-value="apiQuery.js"><a href="apiQuery.js.html">apiQuery.js</a></td>
	<td data-value="86.48" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 86%"></div><div class="cover-empty" style="width: 14%"></div></div>
	</td>
	<td data-value="86.48" class="pct high">86.48%</td>
	<td data-value="37" class="abs high">32/37</td>
	<td data-value="83.33" class="pct high">83.33%</td>
	<td data-value="6" class="abs high">5/6</td>
	<td data-value="90" class="pct high">90%</td>
	<td data-value="10" class="abs high">9/10</td>
	<td data-value="85.71" class="pct high">85.71%</td>
	<td data-value="35" class="abs high">30/35</td>
	</tr>

<tr>
	<td class="file low" data-value="auth.js"><a href="auth.js.html">auth.js</a></td>
	<td data-value="5.55" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 5%"></div><div class="cover-empty" style="width: 95%"></div></div>
	</td>
	<td data-value="5.55" class="pct low">5.55%</td>
	<td data-value="36" class="abs low">2/36</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="14" class="abs low">0/14</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="8" class="abs low">0/8</td>
	<td data-value="5.55" class="pct low">5.55%</td>
	<td data-value="36" class="abs low">2/36</td>
	</tr>

<tr>
	<td class="file low" data-value="chartDataFormatter.js"><a href="chartDataFormatter.js.html">chartDataFormatter.js</a></td>
	<td data-value="5.26" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 5%"></div><div class="cover-empty" style="width: 95%"></div></div>
	</td>
	<td data-value="5.26" class="pct low">5.26%</td>
	<td data-value="19" class="abs low">1/19</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="9" class="abs low">0/9</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="5.26" class="pct low">5.26%</td>
	<td data-value="19" class="abs low">1/19</td>
	</tr>

<tr>
	<td class="file low" data-value="firebaseQuery.js"><a href="firebaseQuery.js.html">firebaseQuery.js</a></td>
	<td data-value="46.34" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 46%"></div><div class="cover-empty" style="width: 54%"></div></div>
	</td>
	<td data-value="46.34" class="pct low">46.34%</td>
	<td data-value="82" class="abs low">38/82</td>
	<td data-value="3.84" class="pct low">3.84%</td>
	<td data-value="26" class="abs low">1/26</td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="15" class="abs medium">9/15</td>
	<td data-value="48.05" class="pct low">48.05%</td>
	<td data-value="77" class="abs low">37/77</td>
	</tr>

<tr>
	<td class="file low" data-value="inputChecker.js"><a href="inputChecker.js.html">inputChecker.js</a></td>
	<td data-value="20.37" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 20%"></div><div class="cover-empty" style="width: 80%"></div></div>
	</td>
	<td data-value="20.37" class="pct low">20.37%</td>
	<td data-value="54" class="abs low">11/54</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="36" class="abs low">0/36</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="9" class="abs low">0/9</td>
	<td data-value="20.37" class="pct low">20.37%</td>
	<td data-value="54" class="abs low">11/54</td>
	</tr>

<tr>
	<td class="file medium" data-value="insulineCalculator.js"><a href="insulineCalculator.js.html">insulineCalculator.js</a></td>
	<td data-value="57.14" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 57%"></div><div class="cover-empty" style="width: 43%"></div></div>
	</td>
	<td data-value="57.14" class="pct medium">57.14%</td>
	<td data-value="21" class="abs medium">12/21</td>
	<td data-value="66.66" class="pct medium">66.66%</td>
	<td data-value="12" class="abs medium">8/12</td>
	<td data-value="40" class="pct low">40%</td>
	<td data-value="5" class="abs low">2/5</td>
	<td data-value="57.14" class="pct medium">57.14%</td>
	<td data-value="21" class="abs medium">12/21</td>
	</tr>

<tr>
	<td class="file medium" data-value="localStoreManager.js"><a href="localStoreManager.js.html">localStoreManager.js</a></td>
	<td data-value="77.77" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 77%"></div><div class="cover-empty" style="width: 23%"></div></div>
	</td>
	<td data-value="77.77" class="pct medium">77.77%</td>
	<td data-value="63" class="abs medium">49/63</td>
	<td data-value="57.14" class="pct medium">57.14%</td>
	<td data-value="14" class="abs medium">8/14</td>
	<td data-value="56.25" class="pct medium">56.25%</td>
	<td data-value="16" class="abs medium">9/16</td>
	<td data-value="77.41" class="pct medium">77.41%</td>
	<td data-value="62" class="abs medium">48/62</td>
	</tr>

<tr>
	<td class="file high" data-value="testingJsons.js"><a href="testingJsons.js.html">testingJsons.js</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at Thu May 26 2022 11:26:03 GMT+0200 (Central European Summer Time)
            </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    