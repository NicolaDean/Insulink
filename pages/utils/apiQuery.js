
import React from 'react';
import * as testingJson from "./testingJsons"

const debug = true;

export const imgUrl = "https://spoonacular.com/cdn/ingredients_100x100/";
const ingredientSearch = "";
const apiKey = "apiKey=916e4c93a5854139b950d4708c25bd80";
const apiRoot = "https://api.spoonacular.com/";
const apiIngr = "food/ingredients/";


/**
 * eg : https://api.spoonacular.com/food/ingredients/{id}/information
 * retrive details of ingredient with id
 * @param {*} id id of the ingredients
 * @returns details of ingredients
 */
export const getIngredientDetails = (id)=>{

    if(debug) return testingJson.foodDetails;

    let query = apiRoot + apiIngr + id +"/information?" +apiKey +"&amount=100&unit=grams";;
    console.log("QUERY: " + query);
    return getJson(query);
}

/**
 * eg : https://api.spoonacular.com/food/ingredients/search
 * get all ingredients with similar name of userInput
 * @param {*} userInput user query atribute
 * @returns a list of ingredients
 */
export const getIngredentList = (userInput) =>{

    if(debug){
        return testingJson.ingredientList;
    }

    let query = apiRoot + apiIngr + "search?query=" + userInput + "&number=4&" + apiKey
    console.log("QUERY: " + query);
    return getJson(query);
}

/**
 * given a nutrientsJson map each nutritions name to its contents/values
 * @param {*} nutrientsJson 
 * @returns a dictionary of nutrients
 */
export const extractNutrients = (nutrientsJson) =>{

    let nutrients = {};

    nutrientsJson.forEach( nut =>{
        nutrients[nut.name] = nut; //Map each nutrient name to its content
    });

    return nutrients;
}

export const extractProperties = (propertiesJson) =>{
    let properties = {};

    propertiesJson.forEach( p =>{
        properties[p.name] = p; //Map each properties name to its content
    })

    return properties;
}
/**
 * Do a fetch operation on the query given in input and return the json response from API
 * @param {*} query API QUERY
 * @returns 
 */
const getJson = async (query) =>{
    
    //Prepare request
    const response = fetch(query);
    //WAITING response from API
    const json = await(await response).json();

    if(json.code = 402){
        console.log(json)
        console.log("FINISHED DAILY API CALL");
    } 
    else console.log(json);
    //Return
    return json;
}