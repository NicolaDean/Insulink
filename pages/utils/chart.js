import React from 'react';
import Svg,{ G,Circle} from 'react-native-svg';
import {View,Text,StyleSheet} from 'react-native'

//TO DELETE (LASCIATA SOLO PER GARANZIA)
export const DonutChart = ({    data,
                                legend = ["A","B","C"],
                                colors = ["#a70000","#2D31FA","#FFD32D"],
                                radius = 40,
                                width = 400,
                                height = 400,
                                strokeWidth = 10,
                            }) =>{


    if (data ==  undefined) 
         return (<Text>No Data Available</Text>);

    console.log(data);
    const circumference = 2 * Math.PI * radius;
    const halfCircle = radius + strokeWidth;

    let tot = 0;
    let percentages = [];
    let offsets = [];
    let rotation = [];

    //Calculate Total
    data.forEach(element => {
        tot = tot + element;
    });
    
    let i=1;
    rotation.push(0);

    //Calculate Offset and rotation of each cirle in the graph
    data.forEach(element =>{
        let p = element/tot;
        percentages.push(p);
        offsets.push(circumference -  circumference*p);
        rotation.push(rotation[i-1] + 360*p);
        i = i+1;
    });


    //console.log("Offset:\t" + offsets);
    //console.log("Rotation:\t" + rotation);
    
    return (
        <View style={styles.container} height={height} width={width}>
         <View style={styles.graphWrapper}>
            <Text>CHART</Text>
            <Svg height={height} width={width} viewBox="0 0 180 180">
             <G style={styles.container} rotation={-90} originX="90" originY="90" >
                <Circle 
                    cx="50%" 
                    cy="50%" 
                    r={radius} 
                    fill="transparent"
                    stroke={colors[0]}
                    strokeWidth={strokeWidth}
                    strokeLinecap="round"
                    strokeOpacity=".1"
                    originX = "90"
                    originY = "90"
                />
                <Circle 
                    cx="50%" 
                    cy="50%" 
                    r={radius} 
                    fill="transparent"
                    stroke={colors[0]}
                    strokeWidth={strokeWidth}
                    strokeLinecap="round"
                    strokeDasharray={circumference}
                    strokeDashoffset={offsets[0]}
                    rotation={rotation[0]}
                    originX = "90"
                    originY = "90"
                />
                <Circle 
                    cx="50%" 
                    cy="50%" 
                    r={radius} 
                    fill="transparent"
                    stroke={colors[1]}
                    strokeWidth={strokeWidth}
                    strokeLinecap="round"
                    strokeDasharray={circumference}
                    strokeDashoffset={offsets[1]}
                    rotation={rotation[1]}
                    originX = "90"
                    originY = "90"
                />
                <Circle 
                    cx="50%" 
                    cy="50%" 
                    r={radius} 
                    fill="transparent"
                    stroke={colors[2]}
                    strokeWidth={strokeWidth}
                    strokeLinecap="round"
                    strokeDasharray={circumference}
                    strokeDashoffset={offsets[2]}
                    rotation={rotation[2]}
                    originX = "90"
                    originY = "90"
                />
                </G >
                <Legend legend={legend}/>
            </Svg>
            
            </View>
        </View>
    );
}

const Legend = ({legend}) =>{

    return (
        <Svg>
            <G style={styles.graphWrapper}>
                <Circle radius="10" fill="pink" /><Text>{legend[0]}</Text>
                <Circle radius="10" fill="pink" /><Text>{legend[1]}</Text>
                <Circle radius="10" fill="pink" /><Text>{legend[2]}</Text>
            </G>
        </Svg>
        
    );
}

const styles = StyleSheet.create({
    container: {
      flex: 1,
      justifyContent: "center",
      alignItems: "center",
    },
    graphWrapper: {
      alignItems: "center",
      justifyContent: "center",
    },
  });