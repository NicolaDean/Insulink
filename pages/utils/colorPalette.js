
export default {
    primary:"",
    secondary:"",
    black:"",
    white:"",
    background:"white"
}