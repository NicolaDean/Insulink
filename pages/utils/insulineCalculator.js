export class InsulineCalculator {
    actualGlycemia;
    targetGlycemia;
    carbo;
    proteins;
    fats;
    CHORatio;
    insulineSensitivity;
    limit = 180;
    totalInsulineDaily;
    weight;
    basal; //backgound insuline daily

    constructor() {
        //TODO
    }

    calc(userInfo) {
        //Set all user parameters or load from file info
    };
    
    pippo () {
        console.log("banana");
        return "Ciao";
    };
    //Dose calculator
    mealDose(actualGlycemia, carbo, CHORatio) {
        return carbo / Math.floor(CHORatio);
    };

    correctionDose(actualGlycemia, targetGlycemia, insulineSensitivity) {
        return (actualGlycemia - targetGlycemia) / insulineSensitivity;
    };

    totalDose(actualGlycemia, carbo, CHORatio, targetGlycemia, insulineSensitivity) {
        var mealDose = mealDose(actualGlycemia, carbo, CHORatio);
        var correctionDose = 0;
        if (actualGlycemia >= limit) {
            correctionDose = correctionDose(actualGlycemia, targetGlycemia, insulineSensitivity);
        }
        return mealDose + correctionDose;
    };

    //Standard prediction funcions (Preferred if inserted from user)
    totalInsulineDaily() {
        totalInsulineDaily = 0, 55 * weight;
    };

    basal(){
        basal = totalInsulineDaily / 2;
    };

    CHORatioCalculate() {
        CHORatio = 500 / totalInsulineDaily;
    };

    correctionFactorCalculate() {
        insulineSensitivity = 1800 / totalInsulineDaily;
    };
    
   
} 

export default InsulineCalculator;