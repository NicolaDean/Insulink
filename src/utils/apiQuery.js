import * as testingJson from "./testingJsons"
import axios from 'axios';
import { networkErrors } from "../constants/registrationSteps";
import { inputChecker } from "./inputChecker";

const debug = true;

export const imgUrl = "https://spoonacular.com/cdn/ingredients_100x100/";
const ingredientSearch = "";
const apiKey = "apiKey=916e4c93a5854139b950d4708c25bd80";
const apiRoot = "https://api.spoonacular.com/";
const apiIngr = "food/ingredients/";


const edamAppId = "21f2be32";
const edamAppKey = "d0e8b5d4eecb5c925ab79b75c85d0583";

/**
 * eg : https://api.spoonacular.com/food/ingredients/{id}/information
 * retrive details of ingredient with id
 * @param {*} id id of the ingredients
 * @returns details of ingredients
 */
export const getIngredientDetails = (id,amount,unit = "grams")=>{

    if(debug) return testingJson.foodDetails;

    let query = apiRoot + apiIngr + id +"/information?" +apiKey +"&amount=100&unit=grams";;
    console.log("QUERY: " + query);
    return getJson(query);
}

/**
 * eg : https://api.spoonacular.com/food/ingredients/search
 * get all ingredients with similar name of userInput
 * @param {*} userInput user query atribute
 * @returns a list of ingredients
 */
export const getIngredentList = (userInput) =>{

    if(debug){
        return testingJson.ingredientList;
    }

    let query = apiRoot + apiIngr + "search?query=" + userInput + "&number=4&" + apiKey
    console.log("QUERY: " + query);
    return getJson(query);
}

/**
 * eg: https://api.spoonacular.com/food/search?query=apple&number=2
 * 
 * @param {*} userInput 
 * @returns list of all foods/recipes/articles/menu item related to query
 */
export const getAllRelatedFood = (userInput) =>{

    if(debug){
        return testingJson.allFoodList;
    }
    
    let query = apiRoot + "food/search?query=" + userInput + "&number=4&" + apiKey;
    console.log("QUERY: " + query);
    return getJson(query);
}
/**
 * given a nutrientsJson map each nutritions name to its contents/values
 * @param {*} nutrientsJson 
 * @returns a dictionary of nutrients
 */
export const extractNutrients = (nutrientsJson) =>{

    let nutrients = {};

    nutrientsJson.forEach( nut =>{
        nutrients[nut.name] = nut; //Map each nutrient name to its content
    });

    return nutrients;
}

export const extractProperties = (propertiesJson) =>{
    let properties = {};

    propertiesJson.forEach( p =>{
        properties[p.name] = p; //Map each properties name to its content
    })

    return properties;
}
/**
 * Do a fetch operation on the query given in input and return the json response from API
 * @param {*} query API QUERY
 * @returns 
 */
const getJson = async (query) =>{
    
    //Prepare request
    const response = fetch(query);
    //WAITING response from API
    const json = await(await response).json();

    if(json.code = 402){
        //console.log(json)
        //console.log("FINISHED DAILY API CALL");
    } 
    //else console.log(json);
    //Return
    return json;
}

//OTHER API TEST: NUTRITIONIX

rootUrl = 'https://trackapi.nutritionix.com/v2';
nutrixAppId = 'e29d5b1a';
nutrixAppKey = '4bfbc69b7095239b9768d1f7b3f47e56';

methods = {
    get:"GET",
    post:"POST"
}

headers = {
    'x-app-id':nutrixAppId,
    'x-app-key':nutrixAppKey,
    'x-remote-user-id': 0
}

class Api{
    
    errorFunc = (error) => { console.log("NETWORK ERROR: " + JSON.stringify(error))};

    constructor(){}

    setErrorFunc = (func) =>{
        this.errorFunc = func;
    }
    
    /**
     * 
     * @param {*} userInput 
     * @param {*} userData 
     * @returns 
     */
    getSportCalories = async (userInput,userData) =>{

        //TODO ADD AGE AND GENDER TO PERSONAL DATA
        const param = {
            "query":userInput,
            "gender":"male",
            "weight_kg":userData.weight,
            "height_cm":userData.height,
            "age":30
           }
    
           return await this.doRequest(methods.post,'/natural/exercise',param);
    
    }

    /**
     * allow to search for foods on the API database, both common and branded foods
     * @param {*} userInput food to search
     * @returns a list of foods
     */
    getFoodListAlternative = async (userInput) =>{

        const param = {
            query: userInput
        }

        return await this.doRequest(methods.get,'/search/instant',param);
    }

    //LINK TO GENERATE UPC 
    //https://docs.google.com/document/d/1hZ69q8BhEgEVHbFzQnPlGNPjyJycKWBDhY1jsT13np8/mobilebasic
    getFoodListBarCode = async (userInput) =>{
     //console.log('upc  '+ typeof userInput+userInput)
        const param = {
            upc: 851045005013
        }

        return await this.doRequest(methods.get,'/search/item',param);
    }    

    /**
     * Allow to retrive details of specific foods
     * @param {*} userInput id/name of food
     * @returns details of food
     */
    getIngredientDetailsAlternative = async  (userInput) =>{
    
        const param = {
            query:userInput,
            num_servings:1
        }

        return await this.doRequest(methods.post,'/natural/nutrients',param);
    }
    /**
     * General purpose function to do API request with axios 
     * @param {*} method method to use (POST,GET)
     * @param {*} query relative path to the API request
     * @param {*} param parameters of request (eg query,foodAmount...)
     * @returns JSON with API response
     */
    doRequest = async(method,query,param) =>{
        console.log("QUERY : " + query + JSON.stringify(param));
        this.errorFunc("");
        const response = await axios({
                url:rootUrl + query,
                method:method,
                headers:headers,//TODO put data into params for GET and in data for POST
                params:(method == methods.get)? param : undefined, 
                data:(method == methods.post)? param : undefined,
                responseType: 'json'
                }).then( (response) => {
            return (response.data);
          })
          .catch( (error)=> {
    
            //this.errorFunc(networkErrors.APIerror);
            console.log(error);
            inputChecker.apiErrorHelper(error,this.errorFunc);

            console.log("BANANA" + JSON.stringify(error));
            //console.log(JSON.stringify(error));
            /*if(error.status == 401) {
                console.log("ENDED STUDENT LIMITS!!!!!!!!!!!!!!!!!!!!");
            }*/
            return (testingJson.foodDetails);
          });
    
          return response;
    }

}

export const Food_API = new Api();