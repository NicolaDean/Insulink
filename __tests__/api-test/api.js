import axios from 'axios';
import React from 'react';
import { api_axios,apiPath, Food_API } from '../../src/utils/apiQuery';
//import MockAdapter from 'axios-mock-adapter'


const x = jest.fn(async(config) =>{
    console.log("CONFIG:" + JSON.stringify(config));
    return new Promise(resolve=>resolve()); //THIS ALLOW TO FAKE THE REQUEST
});
let mock_axios;
jest.mock('axios', () => mock_axios = jest.fn(async(config) =>{
    console.log("CONFIGg:" + JSON.stringify(config));
    return new Promise(resolve=>resolve()); //THIS ALLOW TO FAKE THE REQUEST
}));


const dummyFoodList = {foods:[{food:"apple"}]};

//TODO ADD A TEST FOR CHECK NETWORK ERROE (eg by putting a bool in mock_axios with "throw error")
//MOCK FOOD_API.onError so we can check if it is called with the error

/**
 * THIS TESTS ALLOW TO CHECK IF AXIOS IS CALLED WITH THE CORRECT CONFIG OBJECT 
 * THE CHECK IS DONE BY SNAPSHOT OF MOCKED CONFIG 
 * (containing parameters, HTTP method, header...)
 */
describe("Food_API component TEST:",()=>{

    beforeAll(()=>{
        mock_axios.mockClear();
    });

    afterEach(() => {
        mock_axios.mockClear(); //TO RESET CALL STACK OF MOCK OBJECT
        //jest.restoreAllMocks();
    });

    test('Do Food Search', async () => {

        const userInput = "apple";

        const res = await Food_API.getFoodListAlternative(userInput);
          
        const axiosConfig = mock_axios.mock.calls[0];
        //I CHECK THAT THE AXIOS API IS CALLED ALWAYS WITH SAME CONFIGURATION
        expect(axiosConfig).toMatchSnapshot();
        
    });

    test('Retrive Food Details', async () => {

        const userInput = "apple";

        const res = await Food_API.getIngredientDetailsAlternative(userInput);
          
        const axiosConfig = mock_axios.mock.calls[0];

        console.log(JSON.stringify(mock_axios.mock.calls));
        //I CHECK THAT THE AXIOS API IS CALLED ALWAYS WITH SAME CONFIGURATION
        expect(axiosConfig).toMatchSnapshot();
    });

    test('Retrive Sport Calories', async () => {

        const userData = {weight:80,height:180};
        const userInput = "run for 2 hours";

        const res = await Food_API.getSportCalories(userInput,userData);
          
        const axiosConfig = mock_axios.mock.calls[0];

        console.log(JSON.stringify(mock_axios.mock.calls));
        //I CHECK THAT THE AXIOS API IS CALLED ALWAYS WITH SAME CONFIGURATION
        expect(axiosConfig).toMatchSnapshot();

    });

    test('Get Barcode Food', async () => {
        const userInput="BARRCODE";
        const res = await Food_API.getFoodListBarCode(userInput);
          
        const axiosConfig = mock_axios.mock.calls[0];

        console.log(JSON.stringify(mock_axios.mock.calls));
        //I CHECK THAT THE AXIOS API IS CALLED ALWAYS WITH SAME CONFIGURATION
        expect(axiosConfig).toMatchSnapshot();
    });
});