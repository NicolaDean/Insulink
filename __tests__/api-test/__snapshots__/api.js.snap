// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Food_API component TEST: Do Food Search 1`] = `
Array [
  Object {
    "data": undefined,
    "headers": Object {
      "x-app-id": "e29d5b1a",
      "x-app-key": "4bfbc69b7095239b9768d1f7b3f47e56",
      "x-remote-user-id": 0,
    },
    "method": "GET",
    "params": Object {
      "query": "apple",
    },
    "responseType": "json",
    "url": "https://trackapi.nutritionix.com/v2/search/instant",
  },
]
`;

exports[`Food_API component TEST: Get Barcode Food 1`] = `
Array [
  Object {
    "data": undefined,
    "headers": Object {
      "x-app-id": "e29d5b1a",
      "x-app-key": "4bfbc69b7095239b9768d1f7b3f47e56",
      "x-remote-user-id": 0,
    },
    "method": "GET",
    "params": Object {
      "upc": "BARRCODE",
    },
    "responseType": "json",
    "url": "https://trackapi.nutritionix.com/v2/search/item",
  },
]
`;

exports[`Food_API component TEST: Retrive Food Details 1`] = `
Array [
  Object {
    "data": Object {
      "num_servings": 1,
      "query": "apple",
    },
    "headers": Object {
      "x-app-id": "e29d5b1a",
      "x-app-key": "4bfbc69b7095239b9768d1f7b3f47e56",
      "x-remote-user-id": 0,
    },
    "method": "POST",
    "params": undefined,
    "responseType": "json",
    "url": "https://trackapi.nutritionix.com/v2/natural/nutrients",
  },
]
`;

exports[`Food_API component TEST: Retrive Sport Calories 1`] = `
Array [
  Object {
    "data": Object {
      "age": 30,
      "gender": "male",
      "height_cm": 180,
      "query": "run for 2 hours",
      "weight_kg": 80,
    },
    "headers": Object {
      "x-app-id": "e29d5b1a",
      "x-app-key": "4bfbc69b7095239b9768d1f7b3f47e56",
      "x-remote-user-id": 0,
    },
    "method": "POST",
    "params": undefined,
    "responseType": "json",
    "url": "https://trackapi.nutritionix.com/v2/natural/exercise",
  },
]
`;
